### PR TITLE
docs(span): finish docs and tests for Span

### PR DESCRIPTION
**Pull request by joshka**
image

### CODE REVIEW

File: src/text/span.rs [hunk 1]
```diff
@@ -6,22 +6,55 @@ use unicode_width::UnicodeWidthStr;
...
+/// style.
+///
+/// It is the smallest unit of text that can be styled. It is used by the [`Line`] type to represent
+/// a line of text where each grapheme has its own style.
+///
+/// # Examples
+///
+/// `Span`a can be created from a `&str`, a `String` or a `Cow<str>`. It can also be created from a
+/// `&str` or a `String` using the `From` trait, which will create a `Span` with no style.
+/// Alternatively, a `Span` can be created from a `&str` or a `String` using the `Stylize` trait,
```
Comment (mindoodoo) [inline src/text/span.rs:19]: All the creation options listed here except for `Cow<str>` seem redundant to the example, so maybe a little wordy considering the example follows right after ?

lmk what you think

File: src/text/span.rs [hunk 2]
```diff
@@ -6,22 +6,55 @@ use unicode_width::UnicodeWidthStr;
...
+/// let span = Span::raw("test content");
+/// let span = Span::raw(String::from("test content"));
+/// let span = Span::from("test content");
+/// let span = Span::from(String::from("test content"));
+/// let span = "test content".green();
+/// let span = String::from("test content").green();
+/// ```
+///
+/// Spans can be styled using the `Stylize` trait to combine foreground and background colors and
+/// modifiers.
```
Comment (mindoodoo) [inline src/text/span.rs:32]:
```suggestion
/// Spans can also be styled using the `Stylize` trait.
```

nit: Imho a bit wordy and doesn't need to detail what can be stylized

File: src/text/span.rs [hunk 3]
```diff
@@ -54,78 +86,72 @@ impl<'a> Span<'a> {
...
+            .graphemes(true)
+            .filter(|g| *g != "\n")
             .map(move |g| StyledGrapheme {
                 symbol: g,
                 style: base_style.patch(self.style),
             })
-            .filter(|s| s.symbol != "\n")
     }
 
     /// Patches the style an existing Span, adding modifiers from the given style.
```
Comment (mindoodoo) [inline src/text/span.rs:130]:
```suggestion
    /// Patches the style of the Span, adding modifiers from the given style.
```

**Review by mindoodoo — APPROVED** [review]
Couple nitpicks otherwise LGTM ! Thanks for this PR, was needed !

**Comment by joshka** [comment]
Updated wording to simplify it somewhat. My reason for being more verbose than necessary is to make it easy on new rust developers (who may not have seen used `Cow`), and older tui-rs developers that may not be aware of the `Stylize` trait and the shortcut methods.

---

Screen Shot 2023-08-24 at 22 49 20
